system/jhtree/ctfile.cpp

```diff
@@ -810,6 +821,33 @@ bool CJHTreeNode::contains(const char *src) const
     return true;
 }
 
+void CJHTreeNode::dump(FILE *out, int length, unsigned rowCount, bool raw) const
+{
+    if (!raw)
+        fprintf(out, "Node dump: fpos(%" I64F "d) type %s\n", getFpos(), getNodeTypeName());
+    if (rowCount==0 || rowCount > getNumKeys())
+        rowCount = getNumKeys();
+    for (unsigned int i=0; i<rowCount; i++)
+    {
+        char *dst = (char *) alloca(getKeyLen()+50);
```

Comment:
+50?  
Also can alloca in a loop be dangerous?  Safer to dynamically allocate e.g. with MemoryAttr or similar.